src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,125 @@
+        registrationService = new RegistrationServiceImpl();
+        storageDao = new StorageDaoImpl();
+    }
+
+    @BeforeEach
+    public void addValidUsers() {
+        Storage.people.clear();
+        firstUser = storageDao.add(new User(null,
+                FIRST_LOGIN, FIRST_VALID_PASSWORD, FIRST_VALID_AGE));
+        secondUser = storageDao.add(new User(null,
+                SECOND_LOGIN, SECOND_VALID_PASSWORD, SECOND_VALID_AGE));
+    }
+
+    @Test
+    public void userWasAddedAndReturned_Ok() {
+        User expected = new User(
+                (long) FIRST_VALID_AGE, THIRD_LOGIN, FIRST_VALID_PASSWORD, FIRST_VALID_AGE);
+        User actual = registrationService.register(expected);
+        assertEquals(expected, actual, "User should be returned");
+    }
+
+    @Test
+    public void registerNullUser_NotOk() {
+        assertThrows(InvalidInputException.class, () ->
+                registrationService.register(null));
+    }
+
+    @Test
+    public void registerNullAge_NotOk() {
+        firstUser.setAge(null);
+        assertThrows(InvalidInputException.class, () ->
+                        registrationService.register(firstUser),
+                "Method should throw InvalidInputException for null age");
+    }
+
+    @Test
+    public void registerNullLogin_NotOk() {
+        firstUser.setLogin(null);
+        assertThrows(InvalidInputException.class, () ->
+                        registrationService.register(firstUser),
+                "Method should throw InvalidInputException for null login");
+    }
+
+    @Test
+    public void registerNullPassword_NotOk() {
+        firstUser.setPassword(null);
+        assertThrows(InvalidInputException.class, () ->
+                        registrationService.register(firstUser),
+                "Method should throw InvalidInputException for null password");
+    }
+
+    @Test
+    public void registerLoginExists_NotOk() {
+        secondUser.setLogin(FIRST_LOGIN);
+        assertThrows(InvalidInputException.class, () -> registrationService.register(secondUser),
+                "Method should throw InvalidInputException if login exists");
+    }
+
+    @Test
+    public void registerUserExists_NotOk() {
+        assertThrows(InvalidInputException.class, () -> registrationService.register(firstUser),
+                "Method should throw InvalidInputException if user exists");
+    }
+
+    @Test
+    public void registerInvalidUsersAge_NotOk() {
+        firstUser.setAge(INVALID_AGE);
+        assertThrows(InvalidInputException.class, () -> registrationService.register(firstUser),
+                "Method should throw InvalidInputException for invalid age");
+
+    }
+
+    @Test
+    public void registerInvalidUsersPassword_NotOk() {
+        firstUser.setPassword(INVALID_PASSWORD);
+        assertThrows(InvalidInputException.class, () -> registrationService.register(firstUser),
+                "Method should throw InvalidInputException for invalid password");
```

Comment:
we can create a constant field with exception name, so when exception class name changes you dont need to rewrite messages in all the tests